packages/devtools_app/lib/src/shared/ui/filter.dart

```diff
@@ -855,6 +855,7 @@ class FilterTag {
 
   static FilterTag? parse(String value) {
     final parts = value.split(filterTagSeparator);
+    if (parts.length < 2) return null;
```

Comment:
Why does the try / catch not work as expected? This seems like a workaround just for our codebase but not a true solution for dart2wasm.

Reply:
dart2wasm release mode omits bounds checks in standard library functions, same as dart2js.

Omitting bounds check means we don't do check that the index is within the bounds of the list and throw an exception if not. Instead we index the underlying Wasm array directly, assuming the index is within the bounds.

Wasm is a "safe" platform, meaning it always does bounds checks itself. But when a bounds check fails it "traps", i.e. crashes in a way that cannot be caught from withing the same Wasm execution.

So for example, with the flag that is implied by `dart compile wasm -O4`, `[1, 2][10]` will fail with a Wasm trap instead of a Dart exception.

I think this flag was a mistake, but dart2js did it first, and now dart2wasm has to do it as well to match the performance of dart2js. Or it needs to do much better in other areas so that overall it will still be faster. Unfortunately both the compiler (dart2wasm + wasm-opt) and engines (V8 and other Wasm engines) are not there yet. So for the time being we decided to implement the same flag in dart2wasm.

We should eventually remove it. I don't know if now is the time to remove it though.